src/Contracts/Database/Connection.php

```diff
@@ -49,7 +49,8 @@ public function selectOne($query, $bindings = []);
      *
      * @param  string  $query
      * @param  array   $bindings
-     * @return object
+     *
+     * @return \PDOStatement|\Gibbon\Database\Result The PDO statement, or an empty Result if query failed.
```

Comment:
Thanks for the quick changes, as you can see I've carved some time out of my day to work on PRs 😄 

This may be more clear, as it will return a Result object on success too.

```suggestion
     * @return \Gibbon\Database\Result The Result object, or an empty Result if query failed.
```

Reply:
If I read the current implementation right, the `\Gibbon\Database\Connection::select` method is using `\Gibbon\Database\Connection::run` underneath.

https://github.com/GibbonEdu/core/blob/2b21872b3fafbc73c253cf254028dfb3e3d06a5c/src/Database/Connection.php#L178-L198

1. The method would return `$this->result`.
2. If there is no database error, `$this->result` would be receiving the return value of `$this->pdo->prepare()`.
3. Object `$this->pdo` here is of the \PDO type. The prepare function will therefore returns `\PDOStatement`. Only when there is a PDO Exception, the method would assign `$this->result` the `new Result()` for return.

Reply:
Yes, you read correct. The difference is in the wiring in the MySqlConnector class, which uses the line `$connection->setAttribute(\PDO::ATTR_STATEMENT_CLASS, array(Result::class));` [here](https://github.com/GibbonEdu/core/blob/v23.0.00/src/Database/MySqlConnector.php#L96) to set the return class to a Result, which is an extended version of the base PDOStatement. This Result class enables us to add some more readable methods, such as `fetchGroupedUnique()` to replace `->fetchAll(PDO::FETCH_GROUP|PDO::FETCH_UNIQUE)`, and to ensure that a `fetchAll()` call will always return an array, rather than null or false, to make iterating over results less cumbersome.

Reply:
This should probably belongs to another PR, but the current code arrangement seems strange. The `Connection` does not depends on `MySqlConnector`. `MySqlConnector` uses `Connection`. So `Connection` should not assume that `PDO` object is initialized by `MySqlConnector::connect`. So I think its structurally better to have `configureEncoding` and `setModes` in the `Connection` class instead.

Reply:
Documentation updated. Please check. Thanks.

Reply:
Hmm. The purpose of the MySqlConnector connector class is to produce a fully configured Connection. The Connection contract is then moderately database agnostic, so that the configuration is separate from the function of the Connection, and ideally opening the possibility of future support for database types.

Reply:
In any case, happy to merge this PR, it's looking good 😄